tests/mypy_test.py

```diff
@@ -98,44 +140,58 @@ def main():
     code = 0
     runs = 0
     for major, minor in versions:
-        roots = libpath(major, minor)
         files = []
         seen = {"__builtin__", "builtins", "typing"}  # Always ignore these.
-        for root in roots:
-            names = os.listdir(root)
-            for name in names:
-                full = os.path.join(root, name)
-                mod, ext = os.path.splitext(name)
+
+        # First add standard library files.
+        if major == 2:
+            root = os.path.join("stdlib", PY2_NAMESPACE)
+            for name in os.listdir(root):
+                mod, _ = os.path.splitext(name)
                 if mod in seen or mod.startswith("."):
                     continue
-                if ext in [".pyi", ".py"]:
-                    if match(full, args, exclude_list):
-                        seen.add(mod)
-                        files.append(full)
-                elif os.path.isfile(os.path.join(full, "__init__.pyi")) or os.path.isfile(os.path.join(full, "__init__.py")):
-                    for r, ds, fs in os.walk(full):
-                        ds.sort()
-                        fs.sort()
-                        for f in fs:
-                            m, x = os.path.splitext(f)
-                            if x in [".pyi", ".py"]:
-                                fn = os.path.join(r, f)
-                                if match(fn, args, exclude_list):
-                                    seen.add(mod)
-                                    files.append(fn)
+                add_files(files, seen, root, name, args, exclude_list)
+        else:
+            supported_versions = parse_versions(os.path.join("stdlib", "VERSIONS"))
+            root = "stdlib"
+            for name in os.listdir(root):
+                if name == PY2_NAMESPACE:
+                    continue
+                mod, _ = os.path.splitext(name)
+                if mod not in supported_versions or supported_versions[mod] > (major, minor):
```

Comment:
Should we print a warning or even fail the test if `mod` is not in `supported_versions`?

Reply:
Yes, I think we should, I was thinking about allowing some unrelated files in that directory (like some scripts etc). But we actually should keep directory structure clean. I will update this and will also add in a separate PR later a test that ensures `stdlib` and `stubs` have strict directory structure.